src/inetaddr.c

```diff
@@ -472,18 +472,11 @@ struct inet_ifaddr *inet_addr_ifa_get_expired(int af, const struct netif_port *d
     struct inet_ifaddr *ifa;
     struct inet_device *idev;
 
-    if (!dev) {
-        dev = inet_addr_get_iface(af, addr);
-        if (!dev)
-            return NULL;
-    }
-
-    idev = dev_get_idev(dev);
-    assert(idev != NULL);
-
     ifa = expired_ifa_lookup(idev, addr, 0, af);
```

Comment:
`idev` is an uninitialized pointer now. I don't think the `expired_ifa_lookup` can succeed.